File: crates/flow/tests/test_data/sample.go
// SPDX-FileCopyrightText: 2025 Knitli Inc. <knitli@knit.li>
// SPDX-License-Identifier: AGPL-3.0-or-later

// Sample Go code for testing ThreadParse functionality
package main

import (
	"errors"
	"fmt"
	"log"
)

// User represents a system user
type User struct {
	ID    uint64
	Name  string
	Email string
}

// Role represents user permissions
type Role int

const (
	Admin Role = iota
	UserRole
	Guest
)

// UserManager manages user operations
type UserManager struct {
	users map[uint64]*User
}

// NewUserManager creates a new user manager
func NewUserManager() *UserManager {
	return &UserManager{
		users: make(map[uint64]*User),
	}
}

// AddUser adds a user to the manager
func (m *UserManager) AddUser(user *User) error {
	if user.Name == "" {
		return errors.New("name cannot be empty")
	}
	m.users[user.ID] = user
	return nil
}

// GetUser retrieves a user by ID
func (m *UserManager) GetUser(userID uint64) (*User, bool) {
	user, ok := m.users[userID]
	return user, ok
}

// CalculateTotal calculates sum of values
func (m *UserManager) CalculateTotal(values []int) int {
	total := 0
	for _, v := range values {
		total += v
	}
	return total
}

// ProcessUser processes user data and returns formatted string
func ProcessUser(user *User) (string, error) {
	if user.Name == "" {
		return "", errors.New("name cannot be empty")
	}
	return fmt.Sprintf("User: %s (%s)", user.Name, user.Email), nil
}

func main() {
	user := &User{
		ID:    1,
		Name:  "Alice",
		Email: "alice@example.com",
	}

	manager := NewUserManager()
	if err := manager.AddUser(user); err != nil {
		log.Fatal(err)
	}

	result, err := ProcessUser(user)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(result)

	numbers := []int{1, 2, 3, 4, 5}
	total := manager.CalculateTotal(numbers)
	fmt.Printf("Total: %d\n", total)
}

File: crates/flow/tests/test_data/sample.ts
// SPDX-FileCopyrightText: 2025 Knitli Inc. <knitli@knit.li>
// SPDX-License-Identifier: AGPL-3.0-or-later

/**
 * Sample TypeScript code for testing ThreadParse functionality
 */

import { EventEmitter } from 'events';
import * as path from 'path';

/**
 * User interface representing a system user
 */
export interface User {
    id: number;
    name: string;
    email: string;
}

/**
 * Role enum for user permissions
 */
export enum Role {
    Admin = "admin",
    User = "user",
    Guest = "guest",
}

/**
 * User manager class for handling user operations
 */
export class UserManager extends EventEmitter {
    private users: Map<number, User>;

    constructor() {
        super();
        this.users = new Map();
    }

    /**
     * Add a user to the manager
     */
    addUser(user: User): void {
        if (!user.name) {
            throw new Error("Name cannot be empty");
        }
        this.users.set(user.id, user);
        this.emit('userAdded', user);
    }

    /**
     * Get a user by ID
     */
    getUser(userId: number): User | undefined {
        return this.users.get(userId);
    }

    /**
     * Calculate total from array of numbers
     */
    calculateTotal(values: number[]): number {
        return values.reduce((sum, val) => sum + val, 0);
    }
}

/**
 * Process user data and return formatted string
 */
export function processUser(user: User): string {
    if (!user.name) {
        throw new Error("Name cannot be empty");
    }
    return `User: ${user.name} (${user.email})`;
}

/**
 * Main function demonstrating usage
 */
function main(): void {
    const user: User = {
        id: 1,
        name: "Alice",
        email: "alice@example.com",
    };

    const manager = new UserManager();
    manager.addUser(user);

    const result = processUser(user);
    console.log(result);

    const numbers = [1, 2, 3, 4, 5];
    const total = manager.calculateTotal(numbers);
    console.log(`Total: ${total}`);
}

main();
